src/isomorphic/classic/element/ReactElement.js

```diff
@@ -208,6 +208,12 @@ ReactElement.createElement = function(type, config, children) {
         'React.createElement(...): Expected props argument to be a plain object. ' +
         'Properties defined in its prototype chain will be ignored.'
       );
+
+      warning(
```

Comment:
This is a very hot path in development. We’d need to compare the runtime before and after with a benchmark that mounts thousands of elements to avoid regressions in DEV.